Frontend/ui-driver/src/Screens/SubscriptionScreen/Controller.purs

```diff
@@ -121,8 +121,8 @@ eval BackPressed state =
   else if state.props.subView == ManagePlan then continue state{props { subView = MyPlan}}
   else if state.props.subView == DuesView then continue state{props { subView = MyPlan}}
   else if state.props.subView == DueDetails then 
-          if state.props.myPlanProps.multiTypeDues then continue state{props { subView = DuesView}}
-          else continue state{props { subView = MyPlan}}
+          if state.props.myPlanProps.multiTypeDues then continue state{props { subView = DuesView}, data{myPlanData{selectedDue = ""}}}
```

Comment:
use this condition only for changing the subview as that's the only parameter changing here.